src/components/TodoList/TodoItem.tsx

```diff
@@ -0,0 +1,56 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { Todo } from '../../types/Todo';
+import { RootState } from '../../app/store';
+import { setCurrentTodo } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo;
+};
+
+const TodoItem = ({ todo }: Props) => {
+  const dispatch = useDispatch();
+  const { id, completed, title } = todo;
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const isTodoChosen = currentTodo?.id === id;
+
+  const handleClick = (newTodo: Todo) => {
+    dispatch(setCurrentTodo(newTodo));
+  };
```

Comment:
The `handleClick` function doesn't need to take a parameter. You already have access to the `todo` object from the component's props. You can use it directly within the `handleClick` function.